cmd/ingress-operator/start.go

```diff
@@ -118,12 +124,52 @@ func start() error {
 		os.Exit(1)
 	}
 
+	// Set up the stop channel for the watcher and operator.
+	stop := make(chan struct{})
+	go func() {
+		<-signals.SetupSignalHandler()
+		stop <- struct{}{}
+	}()
+
+	// Set up and start the file watcher.
+	go func() {
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			log.Error(err, "failed to create watcher")
+			os.Exit(1)
+		}
+		defer watcher.Close()
+		if err := watcher.Add(defaultTrustedCABundle); err != nil {
+			log.Error(err, "failed to add file to watcher", "filename", defaultTrustedCABundle)
+			os.Exit(1)
+		}
+		changed := make(chan bool)
+		go func() {
+			orig, err := ioutil.ReadFile(defaultTrustedCABundle)
+			if err != nil {
+				log.Error(err, "failed to read watched file", "filename", defaultTrustedCABundle)
+				os.Exit(1)
+			}
+			for {
+				<-watcher.Events
+				latest, _ := ioutil.ReadFile(defaultTrustedCABundle)
```

Comment:
We should check the error value.

Reply:
err... I missed this. I'll update the pr.